lib/lcevc/lcevc_dil.js

```diff
@@ -0,0 +1,209 @@
+      }
+    }
+  }
+
+  /**
+   * Close LCEVC Dil.
+   * @override
+   * @export
+   */
+  release() {
+    this.hideCanvas();
+    if (this.dil_) {
+      this.dil_.close();
+      this.dil_ = null;
+    }
+    if (this.lcevcContainer_) {
+      this.lcevcContainer_.remove();
+    }
+    if (this.resizeObserver_) {
+      this.resizeObserver_.disconnect();
+      this.resizeObserver_ = null;
+    }
+  }
+
+  /**
+   * Check if the LCEVC Dil lib is present and is supported by the browser.
+   * @return {boolean}
+   * @private
+   */
+  isSupported_() {
+    if (typeof libDPIModule === 'undefined') {
+      shaka.log.alwaysWarn(
+          'Could not Find LCEVC Library dependencies on this page');
+    }
+
+    if (typeof LcevcDil === 'undefined') {
+      shaka.log.alwaysWarn('Could not Find LCEVC Library on this page');
+    } else {
+      if (!LcevcDil.SupportObject.SupportStatus) {
+        shaka.log.alwaysWarn(LcevcDil.SupportObject.SupportError);
+      }
+    }
+
+    return typeof LcevcDil !== 'undefined' &&
+      typeof libDPIModule !== 'undefined' &&
+      LcevcDil.SupportObject.SupportStatus;
+  }
+
+  /**
+   * Get canvas for LCEVC Dil to render if not exists.
+   * @private
+   */
+  getCanvas_() {
+    this.canvas_ = document.getElementsByClassName('shaka-canvas-container')[0];
+    if (LcevcDil.SupportObject.webGLSupport(this.canvas_)) {
+      if (this.canvas_) {
+        this.canvas_.classList.remove('shaka-hidden');
+      }
+    }
+  }
+
+  /**
+   * Updates LCEVC DIL with the index of the variant that is being downloaded.
+   * @param {number} level Index of currently active variant.
+   * @param {shaka.lcevc.Dil.ContainerFormat} containerFormat Container
+   * Format of the stream.
+   * @private
+   */
+  updateLevel_(level, containerFormat) {
+    if (this.dil_) {
+      this.level_ = level;
+      this.containerFormat_ = containerFormat;
+      this.dil_.setLevelSwitching(level,
+          shaka.lcevc.Dil.AutoBuffer.BUFFER_PLAYOUT);
+      this.dil_.setContainerFormat(containerFormat);
+    }
+  }
+
+  /**
+   * Update current active variant
+   * @param {shaka.extern.Track} track
+   */
+  updateVariant(track) {
+    let containerFormat = shaka.lcevc.Dil.ContainerFormat.MPEG2_TS;
+    switch (track.mimeType) {
+      case 'video/webm': {
+        containerFormat = shaka.lcevc.Dil.ContainerFormat.WEBM;
+        break;
+      }
+      case 'video/mp4': {
+        containerFormat = shaka.lcevc.Dil.ContainerFormat.MP4;
+        break;
+      }
+    }
+    this.updateLevel_(track.id, containerFormat);
+  }
+};
+
+/**
+ * Container Formats.
+ * @const @enum {number}
+ */
+shaka.lcevc.Dil.ContainerFormat = {
+  MPEG2_TS: 0,
+  MP4: 2,
+  WEBM: 1,
+};
+
+/**
+ * AutoBuffer - BUFFER_PLAYOUT is lcevcDil mode that switches
+ * variant when the downloaded buffer has finished playing and
+ * buffers from the new variant starts to play.
+ * @const @enum {number}
+ */
+shaka.lcevc.Dil.AutoBuffer = {
+  BUFFER_PLAYOUT: 1,
```

Comment:
As with ContainerFormat, should this be in 'externs/lcevc.js'? Also, can we include an enum for the '0' case to clarify what the behaviour is when this mode is disabled.

Reply:
This is removed and moved to boolean type